src/gui/sdlmain.cpp

```diff
@@ -2815,9 +2828,11 @@ int main(int argc, char* argv[]) {
 			Cross::CreatePlatformConfigDir(config_path);
 			Cross::GetPlatformConfigName(config_file);
 			config_combined = config_path + config_file;
-			if(control->PrintConfig(config_combined.c_str())) {
-				LOG_MSG("CONFIG: Generating default configuration.\nWriting it to %s",config_combined.c_str());
-				//Load them as well. Makes relative paths much easier
+			if (control->PrintConfig(config_combined.c_str())) {
```

Comment:
I think we can make `PrintConfig()`  accept a `const std::string &`, given the users of this function actually want to pass it a `std::string` in the first place, but then have to convert over to `.c_str()`:

```
src/gui/sdlmain.cpp:    if(!f && !control->PrintConfig(path.c_str())) {
src/gui/sdlmain.cpp:    if(!f && !control->PrintConfig(path.c_str())) {
src/gui/sdlmain.cpp:                    if(control->PrintConfig(config_combined.c_str())) {
src/gui/sdlmain.cpp:            if(control->PrintConfig(config_combined.c_str())) {
src/misc/programs.cpp:          if (!control->PrintConfig(name.c_str())) {
```

I think it's a quick a low-risk cleanup that could be tossed in.

Reply:
Good call! Implemented :)